cpp/community_detection_module/community_detection_module.cpp

```diff
@@ -0,0 +1,101 @@
+#include <mg_exceptions.hpp>
+#include <mg_utils.hpp>
+
+#include "algorithm/louvain.hpp"
+
+namespace {
+
+const char *kProcedureGet = "get";
+
+const char *kFieldNode = "node";
+const char *kFieldCommunity = "community_id";
+
+const char *kArgumentWeight = "weight";
+const char *kArgumentColoring = "coloring";
+const char *kArgumentMinGraphShrink = "min_graph_shrink";
+const char *kArgumentCommunityAlgThreshold = "community_alg_threshold";
+const char *kArgumentColoringAlgThreshold = "coloring_alg_threshold";
+
+const double kDefaultWeight = 1.0;
+
+void InsertLouvainRecord(mgp_graph *graph, mgp_result *result, mgp_memory *memory, const std::uint64_t node_id,
+                         const std::uint64_t community) {
+  mgp_result_record *record = mgp::result_new_record(result);
+  if (record == nullptr) {
+    throw mg_exception::NotEnoughMemoryException();
+  }
+
+  mg_utility::InsertNodeValueResult(graph, record, kFieldNode, node_id, memory);
+  mg_utility::InsertIntValueResult(record, kFieldCommunity, community, memory);
+}
+
+void LouvainCommunityDetection(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
+  try {
+    // Get the query module arguments
+    auto maybe_weight = mgp::list_at(args, 0);
+    auto coloring = mgp::value_get_bool(mgp::list_at(args, 1));
+    auto min_graph_shrink = mgp::value_get_int(mgp::list_at(args, 2));
+    auto community_alg_threshold = mgp::value_get_double(mgp::list_at(args, 3));
+    auto coloring_alg_threshold = mgp::value_get_double(mgp::list_at(args, 4));
+
+    auto weight_property = !mgp::value_is_null(maybe_weight) ? mgp::value_get_string(maybe_weight) : "weight";
```

Comment:
What case would `maybe_weight` be NULL in?

Reply:
Forgot to change the implementation after changing documentation. I figured out that the cleanest solution is to set that property default to `weight`.

Reply:
Addressed the change